File: core/src/dc/targetman/TargetmanGame.java
package dc.targetman;

import com.badlogic.gdx.ApplicationAdapter;
import com.badlogic.gdx.Screen;
import com.badlogic.gdx.graphics.g2d.PolygonSpriteBatch;
import com.badlogic.gdx.graphics.glutils.ShapeRenderer;

import dc.targetman.level.LevelController;
import dc.targetman.screens.LevelScreen;
import dclib.graphics.ScreenUtils;
import dclib.graphics.TextureCache;
import dclib.system.ScreenManager;

public class TargetmanGame extends ApplicationAdapter {

	private final ScreenManager screenManager = new ScreenManager();
	private TextureCache textureCache;
	private PolygonSpriteBatch spriteBatch;
	private ShapeRenderer shapeRenderer;

	@Override
	public void create () {
		textureCache = createTextureCache();
		spriteBatch = new PolygonSpriteBatch();
		shapeRenderer = new ShapeRenderer();
		screenManager.add(createLevelScreen());
	}

	@Override
	public void render () {
		ScreenUtils.clear();
		screenManager.render();
	}

	@Override
	public void dispose () {
		textureCache.dispose();
		spriteBatch.dispose();
		shapeRenderer.dispose();
	}

	private Screen createLevelScreen() {
		LevelController controller = new LevelController(textureCache, spriteBatch, shapeRenderer);
		return new LevelScreen(controller);
	}

	private TextureCache createTextureCache() {
		TextureCache textureCache = new TextureCache();
		textureCache.addTexturesAsAtlas("textures/objects/", "objects");
		return textureCache;
	}

}

File: core/src/main/dc/targetman/TargetmanGame.kt
package dc.targetman

import com.badlogic.gdx.ApplicationAdapter
import dc.targetman.command.CommandProcessor
import dc.targetman.screens.ConsoleScreen
import dc.targetman.screens.LevelScreen
import dc.targetman.util.Json
import dclib.graphics.Render
import dclib.graphics.RenderUtils
import dclib.graphics.TextureCache
import dclib.system.ScreenManager
import dclib.ui.UiPack

class TargetmanGame : ApplicationAdapter() {
	private val config = Json.toObject<AppConfig>("config.json")
	private val commandProcessor = CommandProcessor()
	private val screenManager = ScreenManager()
	private lateinit var textureCache: TextureCache
	private lateinit var render: Render
	private lateinit var uiPack: UiPack
	
	override fun create() {
		textureCache = createTextureCache()
		render = Render()
		uiPack = UiPack("ui/test/uiskin.json", "ui/ocr/ocr_32.fnt", "ui/ocr/ocr_24.fnt")
		val consoleScreen = ConsoleScreen(commandProcessor, uiPack)
		val levelScreen = LevelScreen(config, commandProcessor, textureCache, uiPack, render)
		link(consoleScreen, levelScreen)
		screenManager.add(levelScreen)
        screenManager.add(consoleScreen, false)
	}

	private fun link(consoleScreen: ConsoleScreen, levelScreen: LevelScreen) {
		consoleScreen.closed.on {
			consoleScreen.hide()
			levelScreen.show()
		}
		levelScreen.paused.on {
			levelScreen.pause()
			consoleScreen.show()
		}
	}

	override fun render() {
		RenderUtils.clear()
		screenManager.render()
	}

    override fun resize(width: Int, height: Int) {
        screenManager.resize(width, height)
    }

	override fun dispose() {
		uiPack.dispose()
		textureCache.dispose()
		render.dispose()
	}

	private fun createTextureCache(): TextureCache {
		val textureCache = TextureCache()
        textureCache.loadTexturesIntoAtlas("textures/objects", "objects")
        textureCache.loadTexturesIntoAtlas("textures/skins/man", "skins/man")
		return textureCache
	}
}